SDA-7166 Update the model for node_pools endpoints

lucasponce opened the pull request:
It's my first PR on this repo.

The existing "node_pools" endpoints are not exposed to the public model, and this is a prerequisite step before we can connect them with the ROSA CLI.

cc @pvasant @tbrisker 

Please, feel free to suggest if this flow is correct or if I should send this PR with other changes in the SDK and/or another repo.

Thanks,
Lucas

ciaranRoche commented on model/clusters_mgmt/v1/aws_node_pool_type.model:
```diff
@@ -0,0 +1,27 @@
+/*
+Copyright (c) 2021 Red Hat, Inc.
```
![what year?](https://media.giphy.com/media/4Z9jjKGJVM9Ck8E4cN/giphy.gif)

ciaranRoche commented on model/clusters_mgmt/v1/aws_node_pool_type.model:
```diff
@@ -0,0 +1,27 @@
+    InstanceProfile String
+
+	// Optional keys and values that the installer will add as tags to all AWS resources it creates
+	Tags [String]String
```
indentation looks off here

ciaranRoche commented on model/clusters_mgmt/v1/node_pool_autoscaling_type.model:
```diff
@@ -0,0 +1,24 @@
+
+// Representation of a autoscaling in a node pool.
+class NodePoolAutoscaling {
+    // The minimum number of replicas for the node pool.
+	Min Integer
+
+    // The maximum number of replicas for the node pool.
+	Max Integer
```
Indentation here too

ciaranRoche commented on model/clusters_mgmt/v1/node_pool_type.model:
```diff
@@ -0,0 +1,44 @@
+	Autoscaling NodePoolAutoscaling
+
+    // Specifies whether health checks should be enabled for machines in the NodePool.
+    AutoRepair Boolean
```
indentation, 

This is the bane of my existance with this repo, the indentation between ide's and this `model` syntax

ciaranRoche commented on model/clusters_mgmt/v1/node_pool_type.model:
```diff
@@ -0,0 +1,44 @@
+
+// Representation of a node pool in a cluster.
+class NodePool {
+	// ID used to identify the cluster that this machinepool is attached to.
```
Is this correct `ID` :thinking:

ciaranRoche commented on model/clusters_mgmt/v1/node_pools_resource.model:
```diff
@@ -0,0 +1,45 @@
+*/
+
+// Manages the collection of node pools of a cluster.
+resource NodePools {
```
I think this might need a locator, normally a root resource for all resources :thinking:

lucasponce commented:
Thanks, @ciaranRoche; I think I've fixed most of the comments. 
Would you like to give it another pass?
Thx.

ciaranRoche commented on model/clusters_mgmt/v1/cluster_type.model:
```diff
@@ -158,6 +158,9 @@ class Cluster {
 	link MachinePools []MachinePool
 
+    // List of node pools on this cluster
+    link NodePools []NodePool
```
out of curiosity what is a node pool? and how is it different to machine pools?

lucasponce replied:
It's a similar concept but for Hosted clusters.
These NodePools are "detached" from the control plane.
From OCM, the original work created different endpoints for them.

But from ROSA CLI we may invoke one or another depending on hosted cluster or not, that would be my next task and I'm enabling this in the model as a prior step (it wasn't updated before).

ciaranRoche replied:
okay makes sense, can we add that to the comments, as the comments are used to generate the docs in the openapi. Be good to call out the difference between node pools and machine pools.

lucasponce commented:
Updated, tested locally with an example like:

```
package main

import (
	"fmt"
	"log"
	"os"

	"github.com/openshift-online/ocm-sdk-go"
	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
)

func main() {
	logger, err := sdk.NewGoLoggerBuilder().
		Debug(true).
		Build()
	if err != nil {
		log.Fatalf("can't build logger: %v\n", err)
	}

	connection, err := sdk.NewConnectionBuilder().
		Logger(logger).
		Tokens(os.Getenv("OFFLINE_ACCESS_TOKEN")).
		URL("http://localhost:9000").
		Insecure(true).
		Build()
	if err != nil {
		log.Fatalf("Can't build connection: %v", err)
	}
	defer connection.Close()

	clusterId := "20756lq9aha19chgodjd8krdtr49q8o6"
	clusterGetResp, err := connection.ClustersMgmt().V1().Clusters().Cluster(clusterId).Get().Send()
	if err != nil {
		log.Fatalf("could not find cluster : %v", err)
	}

	fmt.Println("got cluster - ", clusterGetResp.Body().Name())

	nodePoolsListResp, err := connection.ClustersMgmt().V1().Clusters().Cluster(clusterId).NodePools().List().Send()
	if err != nil {
		log.Fatalf("could not find node pools for a cluster : %v", err)
	}

	fmt.Println("got list - ", nodePoolsListResp.Items().Len())

	nodePool, err := v1.NewNodePool().Replicas(3).Build()
	if err != nil {
		log.Fatalf("could not build nodepool : %v", err)
	}
	nodePoolAddResp, err := connection.ClustersMgmt().V1().Clusters().Cluster(clusterId).NodePools().Add().Body(nodePool).Send()
	if err != nil {
		log.Fatalf("could not add a node pool for a cluster : %v", err)
	}

	fmt.Println("got new nodepool - ", nodePoolAddResp.Body().HREF())
}
```

Once that the SDK is enabled and updated from main CS code, it can enable more end-to-end tests on this area.

lucasponce commented:
I have a local build of the "ocm-sdk-go" that works locally with the rosa CLI changes.
Once this PR is reviewed+merged, should I update the "ocm-sdk-go"?
How are the updates of the new "ocm-sdk-go" version propagated into ROSA or any other client? Is that done automatically?

pvasant commented on model/clusters_mgmt/v1/node_pool_autoscaling_type.model:
```diff
@@ -0,0 +1,24 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
```
year:)

lucasponce replied:
+1

tbrisker commented:
@lucasponce once this is merged, a new release with it must be made, then the sdk needs to be updated to use the new version, regenerated, and released, and then any consumer can update the version they use. (yes, i know, this is way too long of a process for every api change)

lucasponce commented on model/clusters_mgmt/v1/node_pool_resource.model:
```diff
@@ -0,0 +1,32 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
```
Bad year

lucasponce commented on model/clusters_mgmt/v1/node_pool_type.model:
```diff
@@ -0,0 +1,44 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
```
bad year

lucasponce commented on model/clusters_mgmt/v1/node_pools_resource.model:
```diff
@@ -0,0 +1,45 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
```
year

lucasponce commented:
Thanks, @tbrisker.
I've updated the PR with the minor fixes reviewed by @ciaranRoche and @pvasant.
I think it's ready for merging.

tbrisker reviewed (approved):
Thanks @lucasponce , i'll tag it for you

tbrisker commented:
https://github.com/openshift-online/ocm-api-model/actions/runs/3558592802
